<!-- 
The MIT License (MIT)
Copyright (c) 2016 GRVC University of Seville

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE. 
-->
<launch>
	<!-- Configurable options -->
	<arg name="robot_name"/> <!-- Robot name defines the namespace for all topics regarding the robot -->
	<arg name="id" default="1"/>
	<arg name="pos_x" default="0.0"/>
	<arg name="pos_y" default="0.0"/>
	<arg name="pos_z" default="0.5"/>
	<arg name="gazebo_ns" default="$(arg robot_name)"/>

	<!-- Spawn a quad in the simulator: send the robot XML to param server, push robot_description to factory and spawn robot in gazebo-->
    <include file="$(find mbzirc_gcs_view)/launch/spawn_quad.launch">
		<arg name="robot_name" value="$(arg robot_name)"/>
        <arg name="id" value="$(arg id)"/>
		<arg name="pos_x" value="$(arg pos_x)"/>
		<arg name="pos_y" value="$(arg pos_y)"/>
		<arg name="pos_z" value="$(arg pos_z)"/>
	</include>

	<!-- Spawn launch hal controller -->
	<node name="grvc_hal_$(arg robot_name)" pkg="grvc_quadrotor_hal" type="hal_node" respawn="false" output="screen"
		args="-hal_ns=/$(arg robot_name)/hal -wp_topic=go_to_wp -gazebo_ns=/$(arg robot_name)"/>
</launch>
